### ISSUE TITLE
Mixing Reform

### ISSUE THREAD

**Issue by MineCake147E** [MineCake147E/MonoAudio#11]
# Backgrounds
+ [`SimpleMixer`](https://github.com/MineCake147E/Shamisen/blob/develop/Shamisen/Filters/Mixing/SimpleMixer.cs) was not simple enough to be named after its features.
  + Supporting addition/removal of audio sources makes it complex
# Idea and Progress
- [x] **`SimpleMixer`** Overhaul: Make SimpleMixer as fast and straightforward as possible
  - [x] Mixes ONLY 2 SOURCES with ~~1:1~~ customizable ratio of volume
    - Scale & mix can be done quickly
  - [x] Drop adding/removing/replacing support
    - We can replace source with [`AudioSocket`](https://github.com/MineCake147E/Shamisen/blob/develop/Shamisen/AudioSocket.cs)
- [ ] Add `net5.0 library` project `Shamisen.Game`
  - [ ] Implement SFX System (on another Issue)
- [ ] #12 
- [ ] Implement **`AdvancedMixer`**
  - [ ] Implement **`Data flow Management`** to maximize the output performance for real-time audio mixing and playback.
    - [ ] Set the `Buffering Strategy` for every input to calculate output
      - [ ] `Aggressive Pre-Buffering` for BGM from storage or network stream
      - [ ] `Realtime` for input from Sound Effect mixer(like SFX system introduced above) for minimal latency
    - [ ] Make whole mixing process multi-threaded
      - [ ] Pre-Read() mixing for BGM using Pairwise Summation
      - [ ] On-Read() mixing for SE Mixers(which is also called in parallel)
